### PR TITLE
Add project index card

### LINKED ISSUE #394: Add project banner according to the new designs.
**Description**

A banner is extensively used to present a project's details. It might be helpful to add it as a component to the vocabulary.
image
 
**Additional context**

The scope of the component is yet to be discussed.

**Want to take up?**

Yes

### PR DESCRIPTION
**Pull request by Dhruvi16**
## Fixes

Fixes #394 by @Dhruvi16 

## Description

Adds project index card to the vocabulary.

## Screenshots

image

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/styles/card.scss [hunk 1]
```diff
@@ -278,5 +279,46 @@ $card-header-shadow: none;
                 }
             }
         }
+        &.project-index {
```
Comment (dhruvkb) [inline src/styles/card.scss:281]: 👍 Appreciate you building the entire component using `@extend`. Helps DRY-ing the codebase. Also kudos for keeping the 4-space indentation (although I would appreciate changing the file to 2-space indentation in a different PR).

File: src/styles/card.scss [hunk 2]
```diff
@@ -15,6 +15,7 @@ $card-header-shadow: none;
             .card-title {
                 text-transform: none;
                 line-height: 1.6rem;
+                font-weight: bold;
```
Comment (dhruvkb) [inline src/styles/card.scss]: This changes the title for all cards. Are you sure this matches Figma?

File: src/stories/cards.stories.mdx [hunk 3]
```diff
@@ -281,3 +281,40 @@ import { figmaConfig } from './helpers'
     </div>
   `}</Story>
 </Preview>
+
+
+<Preview>
+  <Story name="Project Index" parameters={{
```
Comment (dhruvkb) [inline src/stories/cards.stories.mdx]:
```suggestion
  <Story name="Project index" parameters={{
```

File: src/stories/cards.stories.mdx [hunk 4]
```diff
@@ -281,3 +281,40 @@ import { figmaConfig } from './helpers'
...
+    design: figmaConfig('776%3A144')
+  }}>{`
+    <div style="width: 400px;">
```
Comment (dhruvkb) [inline src/stories/cards.stories.mdx]: I'd rather see `<style>...</style>` in the story than `style="..."`.
```suggestion
    <style>
      .project-index {
        width: 400px;
      }
    </style>  
    <div class="project-index">
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: src/styles/card.scss (hunk 2 above)
Comment (Dhruvi16) [reply src/styles/card.scss]: image
image

Yes I am sure.

Comment (Dhruvi16) [reply src/styles/card.scss]: But I guess I should do that in a different PR, right?

**Review by hugosolar — APPROVED** [review]

File: src/styles/card.scss (hunk 2 above)
Comment (dhruvkb) [reply src/styles/card.scss]: No, this is fine.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
The 'Visit site' link seems a little misaligned with the other content. Admittedly it's a problem with the 'Button' component so you could choose to solve it separately but please open an issue about this.

Screenshot 2020-06-13 at 11 36 15 PM

File: src/styles/card.scss [hunk 5]
```diff
@@ -278,6 +278,47 @@ $card-header-shadow: none;
...
+        &.project-index {
+            .card-content {
+                .site-link {
+                    @extend .margin-top-small;
+
+                    span {
+                        @extend .has-text-forest-green;
+                    }
+
+                    i {
```
Comment (dhruvkb) [inline src/styles/card.scss]: If the element has a class, use that as the selector.

Comment (dhruvkb) [reply src/styles/card.scss]: Applies to all element selectors in the PR.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**